From 8380e657512aa51d487e36f23431c07307ccc459 Mon Sep 17 00:00:00 2001
From: Jiani Liu <quic_jianil@quicinc.com>
Date: Fri, 7 Apr 2023 10:37:06 +0800
Subject: [PATCH] mac80211: Upgrade the package to 20161201

This patch removes some of the earlier patches which have
got merged in the upstream.

Change-Id: Ic0f977ec76863b53c323c641f6376794198e68d3
Signed-off-by: Tamizh chelvam <tamizhchelvam@codeaurora.org>

Signed-off-by: Jiani Liu <quic_jianil@quicinc.com>

diff --git a/drivers/net/wireless/ath/ath10k/core.c b/drivers/net/wireless/ath/ath10k/core.c
index 2f9be18..16c9b58 100644
--- a/drivers/net/wireless/ath/ath10k/core.c
+++ b/drivers/net/wireless/ath/ath10k/core.c
@@ -3345,6 +3345,16 @@ int ath10k_core_register(struct ath10k *ar,
 
 	queue_work(ar->workqueue, &ar->register_work);
 
+	/* OpenWrt requires all PHYs to be initialized to create the
+	 * configuration files during bootup. ath10k violates this
+	 * because it delays the creation of the PHY to a not well defined
+	 * point in the future.
+	 *
+	 * Forcing the work to be done immediately works around this problem
+	 * but may also delay the boot when firmware images cannot be found.
+	 */
+	flush_workqueue(ar->workqueue);
+
 	return 0;
 }
 EXPORT_SYMBOL(ath10k_core_register);
-- 
2.7.4

